### PR TITLE
Registration system

### PR DESCRIPTION
**Pull request by JoelForamitti**
Corresponding to https://github.com/MILA-Wien/collectivo/pull/121

### CODE REVIEW

File: src/components/datatable/ObjectHistory.vue [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      closeModal();
+    })
+    .catch((error: any) => {
+      errorToast(toast, error);
+    });
+}
+
+const callRevert = (data: any) => {
+  confirm.require({
+    message: `Are you sure you want to revert ${props.schema.label} ${data.data.id} to the version from ${data.data.history_date}?`,
```
Comment (nanu-c) [inline src/components/datatable/ObjectHistory.vue:60]: Translation?
`${t("Are you sure you want to revert")} ${props.schema.label} ${data.data.id} ${t("to the version from")}  ${data.data.history_date}?`,

File: src/components/datatable/ObjectHistory.vue [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    header: "Confirmation",
```
Comment (nanu-c) [inline src/components/datatable/ObjectHistory.vue:61]: Please translate

File: src/components/datatable/ObjectTableFrame.vue [hunk 3]
```diff
@@ -152,18 +172,10 @@ function sendEmails() {
...
-          <PrimeButton
-            icon="pi pi-plus"
-            class="c-icon-button p-button-sm"
-            @click="createObjectFn()"
-          >
-          </PrimeButton>
-        </div>
-        <div class="m-1 text-left">
+    <div class="c-datatable-toolbar">
+      <!-- <template #start> -->
```
Comment (nanu-c) [inline src/components/datatable/ObjectTableFrame.vue:176]: please remove the debug line

File: src/components/memberships/MembershipsRegisterEditor.vue [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      let i = 0;
+      const x = props.schema.fields["status"].choices[status];
+      options.push({
+        label: x,
+        value: status,
+        key: ++i,
+      });
+    }
+  } else {
+    console.log("No choices for status");
```
Comment (nanu-c) [inline src/components/memberships/MembershipsRegisterEditor.vue:131]: log should be removed or more contextualized.

**Review by nanu-c — CHANGES_REQUESTED** [review]

File: src/components/datatable/ObjectBulkEdit.vue [hunk 5]
```diff
@@ -1,86 +1,96 @@
...
+    return;
+  }
+  member[field].push(object);
+}
+
+function removeObject(member: any, field: any, object: any) {
+  object = Number(object);
+  member[field] = member[field].filter((item: any) => item != object);
+}
+
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply src/components/datatable/ObjectBulkEdit.vue:75]: done

File: src/components/datatable/ObjectBulkEdit.vue [hunk 6]
```diff
@@ -90,38 +100,20 @@ function getActions(input_type: any): (keyof typeof actions)[] {
...
 }
 
 async function save() {
   isSaving.value = true;
   try {
-    const promises: any[] = [];
     const actionFn = actions[chosenAction.value!];
-    for (const member of members_temp.value) {
+    for (const member of objects_temp.value) {
       actionFn(member, chosenColumn.value.field, chosenContent.value);
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply src/components/datatable/ObjectBulkEdit.vue:111]: done

File: src/components/datatable/ObjectBulkEdit.vue [hunk 7]
```diff
@@ -90,38 +100,20 @@ function getActions(input_type: any): (keyof typeof actions)[] {
...
-      promises.push(mainStore.update("profilesProfiles", member, member.id));
     }
-    await Promise.all(promises);
-    successToast();
+    await props.store.updateBulk(props.name, objects_temp.value);
+    successToast(toast, "Object has been updated.");
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply src/components/datatable/ObjectBulkEdit.vue:114]: done

File: src/components/datatable/ObjectField.vue [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+      @update:modelValue="(val:any) => $emit('update:modelValue', val)"
+      :disabled="disabled"
+      :class="{ 'p-invalid': isInvalid }"
+      class="flex-none mr-2"
+    />
+    <small
+      v-if="field.help_text"
+      id="user-attr-{{value}}-help"
+      class="p-info"
+      >{{ field.help_text }}</small
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply src/components/datatable/ObjectField.vue:109]: done

File: src/components/datatable/ObjectField.vue [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+        v-if="modelValue"
+        type="button"
+        value="Remove image"
+        @click="removeUpload(name)"
+      />
+    </div>
+    <img
+      :src="pending_uploads[name] ? pending_uploads[name] : modelValue"
+      v-if="modelValue || pending_uploads[name]"
+      alt="Project logo"
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply src/components/datatable/ObjectField.vue:160]: true! :D

File: src/components/datatable/ObjectHistory.vue (hunk 1 above)
Comment (JoelForamitti) [reply src/components/datatable/ObjectHistory.vue:60]: done!

File: src/components/datatable/ObjectTableFrame.vue (hunk 3 above)
Comment (JoelForamitti) [reply src/components/datatable/ObjectTableFrame.vue:176]: done

File: src/components/memberships/MembershipsRegister.vue [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+  router.push(getRoute(event.step - 1));
+};
+const isSaving = ref(false);
+
+async function complete(event: any) {
+  submitData.value[event.name] = createSubmitData(event.data);
+  isSaving.value = true;
+  try {
+    await mainStore.create("membershipsRegister", submitData.value);
+    successToast(toast, "Registration successful.");
```
[the remark this answers is not in the document]
Comment (JoelForamitti) [reply src/components/memberships/MembershipsRegister.vue:95]: done

File: src/components/memberships/MembershipsRegisterEditor.vue (hunk 4 above)
Comment (JoelForamitti) [reply src/components/memberships/MembershipsRegisterEditor.vue:131]: done!

**Review by nanu-c — APPROVED** [review]

**Pull request merged**